GoogleForCreators/web-stories-wp
# Accessibility: Resolve issue with duplicate aria-label in reused color component

## Context

Duplicate landmarks were flagged in e2e now that floating menus are a thing (see https://github.com/GoogleForCreators/web-stories-wp/runs/5647019338?check_suite_focus=true as an example) 

## Summary

The color input is shared across the floating menus and the style panel. When the style panel is open at the same time as the floating menu the label appears duplicate at the container level causing a (moderate) accessibility violation. 

I think there's value in this rule and don't want to turn it off just because of 1 shared input that can be visible in two places at once. My solution here is to make the container label that prefixes the colon of the label (formerly just `Color input: color`) so that the location of the color input could be put in instead. So now we have 2 variants: `Color input: {label}` (default) and `Color input (floating menu): {label}` - could easily have a third that is anything else `super awesome color input : {label}`. Anyway, you get the point. This also keeps all our existing test infrastructure intact. 

## Relevant Technical Choices

Pass in an optional containerLabelBase to distinguish between which variant is in use and avoid duplication


## To-do

So far as I can tell running aXe locally on the editor, this is the only reused component getting caught in this issue. When e2e runs, check to see if there are others. 

Need to update some karma fixture labels

## User-facing changes

Just label changes.

## Testing Instructions

If you have aXe devtools you can run a comparison on main vs this branch. Insert an element into the canvas and with it selected open the style panel for it. Make sure that `text` tab is open and then run aXe on the whole page. 

| main | this branch | 
| --- | --- | 
| <img width="627" alt="Screen Shot 2022-03-30 at 3 23 07 PM" src="https://user-images.githubusercontent.com/10720454/160941640-d79c5a06-0260-4fd6-9430-ab0b17c7e938.png"> | <img width="617" alt="Screen Shot 2022-03-30 at 3 22 13 PM" src="https://user-images.githubusercontent.com/10720454/160941656-df021bda-14a7-471a-8aaa-91b358553ed7.png"> |

This is the issue we're addressing here: 
<img width="595" alt="Screen Shot 2022-03-30 at 3 23 21 PM" src="https://user-images.githubusercontent.com/10720454/160941696-56f9007a-b23c-4282-81d0-c7e1ba5f3166.png">

Alternatively, check and see that the color containers are labeled like so: 
| design panel | floating menu | 
| --- | --- | 
| <img width="545" alt="Screen Shot 2022-03-30 at 3 33 15 PM" src="https://user-images.githubusercontent.com/10720454/160942041-9a7b3384-4baf-444c-bedc-734ce958a8f1.png"> | <img width="583" alt="Screen Shot 2022-03-30 at 3 32 58 PM" src="https://user-images.githubusercontent.com/10720454/160942053-072badec-9dfc-4940-8a0f-385518b130e2.png"> |
| <img width="547" alt="Screen Shot 2022-03-30 at 3 31 28 PM" src="https://user-images.githubusercontent.com/10720454/160942088-e03a1801-208f-4b8e-a643-e1294721b2ba.png"> | <img width="597" alt="Screen Shot 2022-03-30 at 3 32 02 PM" src="https://user-images.githubusercontent.com/10720454/160942069-5a8dd85e-54bd-49ae-acd4-b7270796d217.png"> |







This PR can be tested by following these steps:

1.


## Reviews

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist




---



Fixes #11028
